shell/platform/darwin/macos/framework/Source/FlutterViewProvider.h

```diff
@@ -0,0 +1,29 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterView.h"
+
+@class FlutterEngine;
+
+/**
+ * @brief A facade over FlutterEngine that allows FlutterEngine's children
```

Comment:
Is there a reason this is so tightly bound to `FlutterEngine`? Usually something like this would be a `@protocol`, and concrete implementations (like one bound to an engine object) would be created--potentially file-scoped--by the thing providing it.

So I would expect `@protocol FlutterViewProvider : NSObject`, and for there to be a private class called something like `EngineViewProvider` that implements this weak referencing delegation, which is an implementation detail.

Reply:
Done.